x/rollapp/keeper/keeper.go

```diff
@@ -124,6 +124,11 @@ func (k Keeper) registerDenomMetadata(ctx sdk.Context, rollapp types.Rollapp) er
 			metadata.DenomUnits[j] = &newDu
 		}
 
+		// validate metadata
+		if validity := metadata.Validate(); validity != nil {
```

Comment:
@hoangdv2429 it's too late to detect the malformed metadata.
This should be validated on rollapp creation as well